coqlspclient/proof_file.py

```diff
@@ -410,15 +471,81 @@ def proofs(self) -> List[ProofTerm]:
         """
         return self.__proofs
 
-    @property
-    def context(self) -> FileContext:
-        """
-        Returns:
-            FileContext: Whole context available in the environment of the Coq file.
-        """
-        return self.coq_file.context
+    def add_step(self, step_text: str, previous_step_index: int):
+        proof, prev = self.__find_prev(self.steps[previous_step_index].ast.range)
+        if prev == -1:
+            self._make_change(self._add_step, step_text, previous_step_index)
+        else:
+            self._make_change(self._add_step, step_text, previous_step_index, True)
```

Comment:
I don't understand the `in_proof` attribute (for both `add_step` and `delete_step`).

Reply:
The `in_proof` attribute is used only to improve performance. When adding or deleting a step we need to check if the step is inside a proof (for now). To do that, we need to check the goals with coq-lsp which is an expensive operation. If we find the step in the ProofFile, we already know that the step is inside a proof, and so we don't need to check it again. That's what the `in_proof` attribute is used for.